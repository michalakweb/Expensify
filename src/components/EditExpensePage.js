import React from 'react';

const EditExpensePage = (props) => {
    console.log(props)
    
    return (
        <div>
            Edit expense component
            <p>Editing the item with id number {props.match.params.id}</p>
        </div>
    )
}

export default EditExpensePage;