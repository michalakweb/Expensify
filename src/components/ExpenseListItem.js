import React from 'react';
import { Link } from 'react-router-dom';


const ExpenseListItem = (props) => (
    <div>
        <Link to={`/edit/${props.id}`}>{props.description}</Link>
        <p>Amount: {props.amount}</p>
        <p>Created at: {props.createdAt}</p>
    </div>
);

export default ExpenseListItem;