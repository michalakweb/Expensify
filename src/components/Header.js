import React from 'react';
import {NavLink} from 'react-router-dom'; 

const Header = () => (
    <header>
        <div>
            <h1>Expensify</h1>
            <NavLink to="/" activeClassName='is-active' exact={true}>Dashboard</NavLink>
            <NavLink to="/create" activeClassName='is-active'>Create Expense</NavLink>
            <NavLink to="/help" activeClassName='is-active'>Help page</NavLink>
        </div>
    </header>
);

export default Header;
