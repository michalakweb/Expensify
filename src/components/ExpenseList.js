import React from 'react';
import { connect } from 'react-redux';
import ExpenseListItem from './ExpenseListItem';
import selectExpenses from '../selectors/expenses';

const ExpenseList = (props) => (
    <div>
        <h1>Expense List</h1>
        {props.expenses.map(el => 
            <ExpenseListItem
                description={el.description}
                amount={el.amount}
                createdAt={el.createdAt}
                key={el.id}
                id={el.id}
            /> 
            )
        }
    </div>
);

const mapStateToProps = (state) => {
    return {
        expenses: selectExpenses(state.expenses, state.filters)
    };
}; 

export default connect(mapStateToProps)(ExpenseList);