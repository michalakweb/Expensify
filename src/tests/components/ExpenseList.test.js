import React from 'react';
import { shallow } from 'enzyme';
import { ExpenseList } from '../../components/ExpenseList';
import moment from 'moment';

const expenses = [{
    id: '0',
    description: 'gum',
    note: '',
    amount: 195,
    createdAt: 0
}, {
    id: '1',
    description: 'rent',
    note: '',
    amount: 122195,
    createdAt: moment(0).subtract(4, 'days').valueOf()
}, {
    id: '2',
    description: 'credit',
    note: '',
    amount: 4500,
    createdAt: moment(0).add(4, 'days').valueOf()
}];

test('should render ExpenseList with mock expenses', () => {
    const wrapper = shallow(<ExpenseList expenses={expenses} />);
    expect(wrapper).toMatchSnapshot();
});

test('should render ExpenseList with no expenses', () => {
    const wrapper = shallow(<ExpenseList expenses={[]} />);
    expect(wrapper).toMatchSnapshot();
});
