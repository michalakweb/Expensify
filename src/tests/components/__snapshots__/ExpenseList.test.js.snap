// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpenseList with mock expenses 1`] = `
<div>
  <ExpenseListItem
    amount={195}
    createdAt={0}
    description="gum"
    id="0"
    key="0"
    note=""
  />
  <ExpenseListItem
    amount={122195}
    createdAt={-345600000}
    description="rent"
    id="1"
    key="1"
    note=""
  />
  <ExpenseListItem
    amount={4500}
    createdAt={345600000}
    description="credit"
    id="2"
    key="2"
    note=""
  />
</div>
`;

exports[`should render ExpenseList with no expenses 1`] = `
<div>
  <p>
    No Expenses
  </p>
</div>
`;
